src/client/datascience/notebook/notebookEditor.ts

```diff
@@ -98,8 +97,14 @@ export class NotebookEditor implements INotebookEditor {
         return getNotebookMetadata(this.document);
     }
     onExecutedCode?: Event<string> | undefined;
-    public getContent() {
-        return this.serializer.serializeNotebookDocument(this.document);
+    public async getContent(): Promise<string> {
+        const cells = generateCellsFromNotebookDocument(this.document, false);
```

Comment:
This code is weird, in other places we save the notebook and get the content from the file system, here we convert the notebook to JSON ourselves.
Why can't we do this in other places as well, instead of getting the user to save the notbook to disck (i.e.have a consisten approach).

Reply:
Because there's no NotebookDocument for webview cases

Reply:
Sorry, misunderstood your question, we should've used editor.getContent() in the other place too

Reply:
This doesn't need to be async